src/net.zig

```diff
@@ -0,0 +1,746 @@
+const c = @import("c.zig").c;
+const errors = @import("errors.zig");
+const std = @import("std");
+
+/// The current major version of SDL_net headers.
+pub const major_version = c.SDL_NET_MAJOR_VERSION;
+/// The current minor version of the SDL_net headers.
+pub const minor_version = c.SDL_NET_MINOR_VERSION;
+/// The current micro (or patchlevel) version of the SDL_net headers.
+pub const micro_version = c.SDL_NET_MICRO_VERSION;
+
+/// This function gets the version of the dynamically linked SDL_net library.
+///
+/// ## Return Value
+/// Returns SDL_net version.
+///
+/// ## Version
+/// This function is available since SDL_net 3.0.0.
+pub fn getVersion() std.SemanticVersion {
+    // TODO use NET_Version()
+    return .{
+        .major = major_version,
+        .minor = minor_version,
+        .patch = micro_version,
+    };
+}
+
+/// Initialize the SDL_net library.
+///
+/// ## Remarks
+/// This must be successfully called once before (almost) any other SDL_net function can be used.
+/// It is safe to call this multiple times; the library will only initialize once, and won't deinitialize until `net.quit()` has been called a matching number of times.
+/// Extra attempts to init report success.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_net 3.0.0.
+pub fn init() Error!void {
+    return errors.wrapCallBool(c.NET_Init());
+}
+
+/// Deinitialize the SDL_net library.
+///
+/// ## Remarks
+/// This must be called when done with the library, probably at the end of your program.
+/// It is safe to call this multiple times; the library will only deinitialize once, when this function is called the same number of times as `net.init()` was successfully called.
+/// Once you have successfully deinitialized the library, it is safe to call `net.init()` to reinitialize it for further use.
+///
+/// ## Thread Safety
+/// It is safe to call this function from any thread.
+///
+/// ## Version
+/// This function is available since SDL_net 3.0.0.
+pub fn quit() void {
+    c.NET_Quit();
+}
+
+/// Error, set by a network operation.
+///
+/// ## Version
+/// This error is provided by zig-sdl3.
+pub const Error = error{
+    SdlNetError,
+    SdlError,
```

Comment:
Does this use the correct `SdlError` and not make a new different `SdlError`? I'm not too familiar with zig errors.